### PR TITLE
fix(a380x): Various fixes pt.2: MFD, pitch trim switch, EFIS sync for LS button

### LINKED ISSUE #8880: LS scale flickering with "Sync EFIS controls between Captain and F/O" ON
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-09-05T19:26:32+00:00",
    "ref": "refs/heads/a380x-experimental",
    "sha": "69fbc8e859df03ab2009fe44e2d707e34c6a6e30",
    "actor": "flogross89",
    "event_name": "manual",
    "pretty_release_name": "a380x-experimental:69fbc8e8",
    "version": "a380x-v0.12.0-a380x-experimental.69fbc8e"
}
```

### Describe the bug

With a single LS p/b on, the LS scale flickers. With both LS p/b on, the LS scale displays normally

### Expected behavior

LS scale displaying normally

### Steps to reproduce

1. Turn on CPT LS scale
2. Notice flickering
3. Turn off CPT LS scale and turn on F/O LS scale
4. Notice flickering

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

Bernardo R

### LINKED ISSUE #8905: FMS: Radio Tuning by Ident Broken
### Aircraft Version

Development

### Build info

```json
master
```

### Describe the bug

The FMS pages implement radio tuning by ident incorrectly, always taking the first result with no duplicate navaid page. This means a random result is picked and tuned.

We should disable this function for release to remove the blocker tag if nobody fixes it.

### Expected behavior

The duplicate navaid selection page should pop up when there are duplicates allowing you to pick the correct one.

### Steps to reproduce

Enter any navaid ident with duplicates in the VOR or ILS boxes.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #8906: FMS: De-Selected Navaid Function Doesn't Work
### Aircraft Version

Development

### Build info

```json
master
```

### Describe the bug

The de-selected navaid function does not work, as it sends the ident of the navaid rather than it's databaseId ("icao" in MSFS facility terms when the functionality was implemented).

If nobody fixes this we should disable the function to remove the blocker tag for release.

### Expected behavior

De-selecting a navaid should cause the navaid to not be selected.

### Steps to reproduce

Enter any navaid in the de-selected navaid boxes and watch it still being selected.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #8915: A380 THS trim switches don't work
### Aircraft Version

Development

### Build info

```json
-
```

### Describe the bug

The THS trim switches on the pedestal are not interactive.

### Expected behavior

I should be able to change the THS trim from the switches in the virtual cockpit.

### Steps to reproduce

1. Load the A380X

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by flogross89**
Fixes #8905 
Fixes #8915 
Fixes #8880 
Fixes #8906 

## Summary of Changes

- Check for valid XPDR code in MFD/SURV/CONTROLS page (don't allow 8 or 9)
- MFD/POSITION/NAVAIDS: Allow input of LS back course, throws format error until B/C approaches are implemented in the A380X
- Systems-Host: Enable linting, no functional changes (ignore for review)
- MFD: Fix navaid deselection
- Add cockpit PITCH TRIM switch
- Fix flickering LS scale when enabling EFIS sync in EFB

## Additional context

Discord username (if different from GitHub): floridude

## Testing instructions

A380X only: 
Please verify that the above four issues have been resolved:

1. Open SD FCTL page, move cockpit PITCH TRIM switches, verify that elevator trim moves
2. Open FMS/POSITION/NAVAIDS, de-select Navaids and verify it works via the ND
3. Open FMS/POSITION/NAVAIDS, tune VOR with duplicate identifier (e.g. FRA), verify that de-duplication window (DUPLICATE NAMES or sth) opens
4. Enable EFIS sync via EFB settings -> Realism, enable LS button on CAPT EFIS CP, verify LS also shows on F/O side

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx [hunk 1]
```diff
@@ -137,7 +138,7 @@ export class MfdFmsPositionNavaids extends FmsPage<MfdFmsPositionNavaidsProps> {
 
     this.deselectedNavaids.forEach((v, i) => {
       if (this.props.fmcService.master?.navaidTuner.deselectedNavaids[i]) {
-        v.set(this.props.fmcService.master.navaidTuner.deselectedNavaids[i]);
+        v.set(this.props.fmcService.master.navaidTuner.deselectedNavaids[i].substring(7).trim());
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx:141]:
```suggestion
        // FIXME pass full navaid objects to deselected navaids so we can get the ident.
        // Taking it from the databaseId is not safe but all we can do for now.
        v.set(this.props.fmcService.master.navaidTuner.deselectedNavaids[i].substring(7).trim());
```

File: fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx [hunk 2]
```diff
@@ -590,7 +565,7 @@ export class MfdFmsPositionNavaids extends FmsPage<MfdFmsPositionNavaidsProps> {
                 <InputField<number>
                   dataEntryFormat={new LsCourseFormat()}
                   dataHandlerDuringValidation={async (v) => {
-                    this.props.fmcService.master?.navaidTuner.setIlsCourse(v || null);
+                    this.props.fmcService.master?.navaidTuner.setIlsCourse(v || null, v && v < 0 ? true : false);
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx:568]: setIlsCourse expects [0, 360].
```suggestion
                    this.props.fmcService.master?.navaidTuner.setIlsCourse(v !== null ? Math.abs(v) : null, v && v < 0 ? true : false);
```

File: fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx [hunk 3]
```diff
@@ -1412,7 +1419,7 @@ export class SquawkFormat implements DataEntryFormat<number> {
     }
 
     const nbr = Number(input);
-    if (!Number.isNaN(nbr) && nbr <= this.maxValue && nbr >= this.minValue) {
+    if (!Number.isNaN(nbr) && nbr <= this.maxValue && nbr >= this.minValue && /^[0-7]{4}$/.test(input)) {
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx:1422]: The min/max checks are redundant with the regex, but not harmful.

File: fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx (hunk 2 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/POSITION/MfdFmsPositionNavaids.tsx:570]: Yeah, it felt dirty but it seems it's the only way we can do it atm.

File: fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx (hunk 3 above)
Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx]: cleaned that up a little bit

**Review by tracernz — APPROVED** [review]

**Comment by Letreivchen** [comment]
Quality Assurance Tester Report

Discord Username  : Letreiv 
Object of testing : #8926  
Aircraft          : A32NX/A380X 
Tier of Testing   : 1
Date              : 17/09/2024 

Testing Process: 
 

- Open SD FCTL page, move cockpit PITCH TRIM switches, verify that elevator trim moves --OK

- Open FMS/POSITION/NAVAIDS, de-select Navaids and verify it works via the ND: Loaded in at EDHI, HAM VOR tuned automatically. Then added HAM on the deselect list, HAM was removed from the ND. Manually tuned CDA, added to deselect list, not removed from ND

- Open FMS/POSITION/NAVAIDS, tune VOR with duplicate identifier (e.g. FRA), verify that de-duplication window (DUPLICATE NAMES opens --OK

- Enable EFIS sync via EFB settings -> Realism, enable LS button on CAPT EFIS CP, verify LS also shows on F/O side --OK

Testing Results: 
Passed

Negatives:

none 

Conclusions: 

Manually tuned navaids are not removed from the ND after added on the deselect list, accurate behavior according to current standpoint

**Pull request merged**